types/hooks.d.ts

```diff
@@ -30,7 +30,7 @@ export interface onRequestHookHandler<
 > {
   (
     request: FastifyRequest<RawServer, RawRequest, RequestGeneric>,
-    reply: FastifyReply<RawServer, RawReply, ContextConfig>,
+    reply: FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>,
```

Comment:
Why do we need both `RawRequest` and `RequestGeneric`? Aren't those two the same?

Reply:
No.`RawRequest` is nodejs request and `RequestGeneric` is user defined types for request body, querystring, params and headers.